src/job-execution/Sas9JobExecutor.ts

```diff
@@ -0,0 +1,108 @@
+import { ServerType } from '@sasjs/utils/types'
+import * as NodeFormData from 'form-data'
+import { ErrorResponse } from '../types/errors'
+import { convertToCSV, isRelativePath } from '../utils'
+import { BaseJobExecutor } from './JobExecutor'
+import { Sas9RequestClient } from '../request/Sas9RequestClient'
+
+export class Sas9JobExecutor extends BaseJobExecutor {
+  private requestClient: Sas9RequestClient
+  constructor(
+    serverUrl: string,
+    serverType: ServerType,
+    private jobsPath: string
+  ) {
+    super(serverUrl, serverType)
+    this.requestClient = new Sas9RequestClient(serverUrl, false)
+  }
+
+  async execute(sasJob: string, data: any, config: any) {
+    const program = isRelativePath(sasJob)
+      ? config.appLoc
+        ? config.appLoc.replace(/\/?$/, '/') + sasJob.replace(/^\//, '')
+        : sasJob
+      : sasJob
+    const jobUri = ''
+    let apiUrl = `${config.serverUrl}${this.jobsPath}?${
+      jobUri.length > 0
+        ? '__program=' + program + '&_job=' + jobUri
+        : '_program=' + program
+    }`
+    apiUrl = `${apiUrl}${
+      config.username && config.password
```

Comment:
should there be a `serverType=='SAS9'` check here also?

Reply:
aah - I see - this whole part only executes in the case of SAS 9.  In that case, the `_job=$jobUri` part is unnecessary, since there are no jobUri's in SAS 9

Reply:
Done!